src/i965_drv_video.c

```diff
@@ -974,12 +974,18 @@ i965_GetConfigAttributes(VADriverContextP ctx,
                     attrib_list[i].value |= VA_RC_CBR;
 
                 if (profile == VAProfileVP8Version0_3 ||
-                    profile == VAProfileVP9Profile0 ||
-                    profile == VAProfileH264ConstrainedBaseline ||
+                    profile == VAProfileVP9Profile0)
+                    attrib_list[i].value |= VA_RC_VBR;
+
+                if (profile == VAProfileH264ConstrainedBaseline ||
                     profile == VAProfileH264Main ||
                     profile == VAProfileH264High ||
-                    profile == VAProfileHEVCMain)
-                    attrib_list[i].value |= VA_RC_VBR;
+                    profile == VAProfileHEVCMain) {
+                    if (IS_GEN9(i965->intel.device_info)) {
+                        attrib_list[i].value |= VA_RC_VBR;
+                        attrib_list[i].value |= VA_RC_MB;
```

Comment:
one use case is error handling for RC input checking in driver. One example is The middle user directly configure RC mode without querying the RC capability. vdenc encoder RC mode depend on the fw ready or not. so the potential issue is here.